### PR TITLE
nomerge: test codeowners

### PR DESCRIPTION
**Pull request by connorsmith256**
**NOMERGE** this is simply a test of the recent changes to CODEOWNERS

### CODE REVIEW

File: flake.nix
Comment (connorsmith256) [inline flake.nix:2]: this comment needs to be resolved before someone can merge

**Review by thomastaylor312 — DISMISSED** [review]

**Comment by rvolosatovs** [comment]
> Merging can be performed automatically with 1 approving review.

@connorsmith256 like I mentioned in #728 , this PR will only be mergeable once I approve it

**Comment by connorsmith256** [comment]
@rvolosatovs and as I mentioned in the issue, I'm testing overlapping rulesets, which should allow org maintainers to always approve and merge 👍

**Review by thomastaylor312 — APPROVED** [review]

**Review by brooksmtownsend — APPROVED** [review]

**Review by rvolosatovs — APPROVED** [review]
So wow, changes much good, likey-likey

**Comment by connorsmith256** [comment]
Okay I got it working as desired through the use of two rulesets:
- `pull request (main) - CODEOWNERS` requires a CODEOWNER approval, but can be bypassed by org maintainers
- `pull request (main) - Review required` "only" requires an approval, but **cannot** be bypassed by anyone

So in aggregate, due to how [rule layering](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets#about-rule-layering) works, this means:
- if the review passes status checks and has a CODEOWNER approval, org maintainers and CODEOWNERS should be able to merge
- if the review passes status checks and has an org maintainer approval, org maintainers only can bypass and merge
- if the review doesn't pass status checks or has no reviews, no one can merge

---

Here's a screenshot for what the UI looks like for me, as an org maintainer, without any review. Note I **cannot** bypass the base checks:
Screenshot 2023-10-06 at 11 47 09

---

And here's a screenshot for what the UI looks like for me, as an org maintainer, with a review from an org maintainer but not a CODEOWNER. Since the base checks are met, I **can** bypass and merge:
Screenshot 2023-10-06 at 11 52 31

---

And here's a screenshot for what the UI looks like for a non-org-maintainer, before a CODEOWNER review. Note they cannot merge:
2023-10-06-20_01_35-screenshot


---

And here's the view for org maintainers and CODEOWNERS, after a CODEOWNER review. Note both can merge at this point:
Screenshot 2023-10-06 at 12 02 46

**Pull request closed without merging**
